### PR TITLE
Add Thaumic Energistics support to the Apimancers drainer

### PR DESCRIPTION
**Pull request by Quantumlyy**
This implementation was derived from an example in the Gadomancy mod.

### CODE REVIEW

**Comment by OneEyeMaker** [comment]
Looks good. But this implementation is not complete. ThE doesn't detect tile entities automatically - you need to manually register class that implements `IAspectStorage`.
Example for Gadomancy's Essentia Condenser: https://github.com/GTNewHorizons/ThaumicEnergistics/blob/master/src/main/java/thaumicenergistics/common/integration/tc/EssentiaTileContainerHelper.java#L387

In other words, you need to call (in post-init phase):
```
// do not copy this code ;)
if (Loader.isModLoaded("thaumicenergistics") {
    ThEApi.instance().transportPermissions().addAspectStorageTileToBothPermissions(TileEntityApimancersDrainerCommon.class);
}
```

File: src/main/java/magicbees/main/Config.java [hunk 1]
```diff
@@ -226,6 +228,15 @@ public void setupItems() {
         setupMiscForestryItemHooks();
     }
 
+    public void setupThaumicEnergistics() {
+        if (isThaumicEnergisticsLoaded) {
+            try {
+                Class c = BlockApimancersDrainer.drainer;
```
Comment (OneEyeMaker) [inline src/main/java/magicbees/main/Config.java:234]: You should register TileEntity class (read method name and check code in my previous comment). And `try`/`catch` block isn't needed here.

Comment (Quantumlyy) [reply src/main/java/magicbees/main/Config.java:234]: > You should register TileEntity class (read method name and check code in my previous comment).
Yes, `BlockApimancersDrainer.drainer` is the used TileEntity as defined on L#522 (see below):
```java
            if (isGTNHCoreModLoaded) BlockApimancersDrainer.drainer = TileEntityApimancersDrainerGT.class;
            apimancersDrainer = new BlockApimancersDrainer();
            GameRegistry.registerBlock(apimancersDrainer, "apimancersDrainer");
            GameRegistry.registerTileEntity(BlockApimancersDrainer.drainer, "apimancersDrainer");
```

Comment (Quantumlyy) [reply src/main/java/magicbees/main/Config.java:234]: > And try/catch block isn't needed here.

This follows the provided example from Gadomancy, and IntelliJs suggestion as apparently `transportPermissions` can be null.

**Review by OneEyeMaker — APPROVED** [review]

**Pull request merged**